control-plane/agents/core/src/core/reconciler/pool/mod.rs

```diff
@@ -0,0 +1,112 @@
+    pub fn from(period: PollPeriods) -> Self {
+        PoolReconciler {
+            counter: PollTimer::from(period),
+        }
+    }
+    /// Return new `Self` with the default period
+    pub fn new() -> Self {
+        Self::from(1)
+    }
+}
+
+#[async_trait::async_trait]
+impl TaskPoller for PoolReconciler {
+    async fn poll(&mut self, context: &PollContext) -> PollResult {
+        let mut results = vec![];
+        for pool in context.registry().specs().get_locked_pools() {
+            if pool.lock().status().created() {
+                results.push(missing_pool_state_reconciler(pool, context).await)
+            }
+        }
+        Self::squash_results(results)
+    }
+
+    async fn poll_timer(&mut self, _context: &PollContext) -> bool {
+        self.counter.poll()
+    }
+}
+
+/// If a pool has a spec but not state, it means that the mayastor instance where the pool should
+/// exist does not have the pool open.
+/// This can happen if the pool is destroyed under the control plane, or if mayastor
+/// crashed/restarted.
+/// In such a case, we issue a new create pool request against the mayastor instance where the pool
+/// should exist.
+#[tracing::instrument(skip(pool_spec, context), fields(pool.uuid = %pool_spec.lock().id))]
+async fn missing_pool_state_reconciler(
+    pool_spec: Arc<Mutex<PoolSpec>>,
+    context: &PollContext,
+) -> PollResult {
+    if !pool_spec.lock().status().created() {
+        // nothing to do here
+        return PollResult::Ok(PollerState::Idle);
+    }
+    let pool_id = pool_spec.lock().id.clone();
+
+    if context.registry().get_pool_state(&pool_id).await.is_err() {
+        let _guard = match pool_spec.operation_guard(OperationMode::ReconcileStart) {
+            Ok(guard) => guard,
+            Err(_) => return PollResult::Ok(PollerState::Busy),
+        };
+        let pool = pool_spec.lock().clone();
+
+        let warn_missing = |pool_spec: &Arc<Mutex<PoolSpec>>, node_status: NodeStatus| {
+            let node_id = pool_spec.lock().node.clone();
+            pool.debug_span(|| {
+                tracing::debug!(
+                    node.uuid = %node_id,
+                    node.status = %node_status.to_string(),
+                    "Attempted to recreate missing pool state, but the node is not online"
+                )
+            });
+        };
+        let node = match context.registry().get_node_wrapper(&pool.node).await {
+            Ok(node) if !node.lock().await.is_online() => {
+                let node_status = node.lock().await.status().clone();
+                warn_missing(&pool_spec, node_status);
+                return PollResult::Ok(PollerState::Idle);
+            }
+            Err(_) => {
+                warn_missing(&pool_spec, NodeStatus::Unknown);
+                return PollResult::Ok(PollerState::Idle);
+            }
+            Ok(node) => node,
+        };
+
+        pool.warn_span(|| tracing::warn!("Attempting to recreate missing pool state"));
```

Comment:
Nit: Perhaps remove "state" from the end. We aren't just recreating the state, we are actually creating the pool.